### ISSUE TITLE
Volume Overlay (Key Press layer conditional override) is delayed by several seconds.

### ISSUE THREAD

**Issue by kathampy** [antonpup/Aurora#1765]
### Expected Behaviour
Volume Overlay (Key Press Override) should respond immediately.

### Actual Behaviour
It takes several seconds of repeated pressing `VolumeUp` / `VolumeDown` before they are recognized by the `Key Press` layer conditional override to enable the `Volume Overlay` layer.

### Reproduction steps
Press the volume keys once, and the overlay does not trigger. Repeatedly press the volume keys for several seconds, and the overlay finally appear. While the overlay is enabled for 3 seconds, all volume changes are instantly recognized.

### Frequency
Every time

### Aurora Version
0.7.0

### Previously an Issue?
No

**Comment by Wibble199** [comment]
Does it behave like this if you were to use other keys?

In the meanwhile, you could also try replacing the two keydown conditions with a "Numeric change detector" and setting the watch value of that to be the `LocalPCInfo\Volume` game state.

**Comment by kathampy** [comment]
I'll test it and report back. I created another solid fill layer below the volume overlay for a dimming effect, and assigned it the same volume key press override. The two layers activate at different times when pressing the volume keys, and sometimes only one activates/

**Comment by kathampy** [comment]
> Does it behave like this if you were to use other keys?
> 
> In the meanwhile, you could also try replacing the two keydown conditions with a "Numeric change detector" and setting the watch value of that to be the `LocalPCInfo\Volume` game state.

Other keys respond instantly. The problem is only with the volume roller. I have a Logitech G810.

The `Numeric State Variable` does not list any expressions in the drop down if it is a child of `Number Change Detector`. It only lists expressions if it's the top level element. I entered `LocalPCInfo/SystemVolume` manually into `default.json` as a workaround. It now detects instantly, but the lighting flickers only when pressing the volume keys as it has no retain duration.

**Comment by Wibble199** [comment]
Yeah there's a known issue with the variables propogating down properly. Sometimes the vars reappear if you select a different property and then go back, but not always. :\

There should be a "True extender" (I think that's what it's called) which does what it's name suggests - extends the length of a true signal. If you put the change detector in one of these (you can use the copy and paste buttons so you don't have to repeat the JSON editing process), you should be able to make it stay on for a few seconds.
